Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/entryDetailsCard.vue

```diff
@@ -0,0 +1,234 @@
+import LaboratoryTimelineComponent from "./laboratory.vue";
+import MedicationTimelineComponent from "./medication.vue";
+import NoteTimelineComponent from "./note.vue";
+
+@Component({
+    components: {
+        LoadingComponent,
+        DatePickerComponent,
+        MedicationComponent: MedicationTimelineComponent,
+        ImmunizationComponent: ImmunizationTimelineComponent,
+        LaboratoryComponent: LaboratoryTimelineComponent,
+        EncounterComponent: EncounterTimelineComponent,
+        NoteComponent: NoteTimelineComponent,
+    },
+})
+export default class EntryDetailsComponent extends Vue {
+    @Getter("user", { namespace: "user" }) user!: User;
+
+    @Getter("isVisible", { namespace: "idle" }) isIdleWarningVisible!: boolean;
+
+    private windowWidth = 0;
+
+    private get isMobileView(): boolean {
+        return this.windowWidth < 576;
+    }
+
+    private entry: TimelineEntry | null = null;
+    private entryDate = "";
+
+    private eventBus = EventBus;
+
+    private isVisible = false;
+
+    private get backButtonIcon(): IconDefinition {
+        return faLongArrowAltLeft;
+    }
+
+    private get modalTitle(): string {
+        return "";
+    }
+
+    private mounted() {
+        this.clear();
+        this.eventBus.$on(EventMessageName.ViewEntryDetails, this.viewDetails);
+    }
+
+    private validations() {
+        return {
+            title: {
+                required: required,
+            },
+        };
+    }
+
+    private isValid(param: Validation): boolean | undefined {
+        return param.$dirty ? !param.$invalid : undefined;
+    }
+
+    private dateString(entryDate: DateWrapper): string {
+        const today = new DateWrapper();
+        if (entryDate.isSame(today, "day")) {
+            return "Today";
+        } else if (entryDate.year() === today.year()) {
+            return entryDate.format("MMM d");
+        } else {
+            return entryDate.format("yyyy-MM-dd");
+        }
+    }
+
+    public viewDetails(entry: TimelineEntry): void {
+        this.clear();
+        this.entry = entry;
+        this.entryDate = this.dateString(entry.date);
+        this.windowWidth = window.innerWidth;
+        this.isVisible = true;
+    }
+
+    private getComponentForEntry(): string {
+        switch (this.entry?.type) {
+            case EntryType.Medication:
+                return "MedicationComponent";
+
+            case EntryType.Immunization:
+                return "ImmunizationComponent";
+
+            case EntryType.Laboratory:
+                return "LaboratoryComponent";
+
+            case EntryType.Encounter:
+                return "EncounterComponent";
+
+            case EntryType.Note:
+                return "NoteComponent";
+            default:
+                return "";
+        }
+    }
+
+    public hideModal(): void {
+        this.$v.$reset();
+        this.isVisible = false;
+        this.clear();
+    }
+
+    private clear() {
+        this.entry = null;
```

Comment:
instead of calling this function to clear the entry, you can just set it to null on the other places and remove this function